subjects/devops/cloud-design/audit/README.md

```diff
@@ -0,0 +1,127 @@
+
+- Can you explain the architecture of your cloud-based microservices application and how the different components interact with each other?
+
+- How did you manage the cost of your solution, and what strategies did you use to optimize it?
+
+- How did you ensure that the application is secure, and what security best practices did you implement to protect your application?
+
+- What monitoring and logging tools did you use, and how did they help you identify and troubleshoot issues with your application?
+
+- Can you explain the auto-scaling policies you implemented and how they allow your application to handle varying workloads?
+
+- How did you optimize the Docker images for each microservice, and how did it impact build times and image sizes?
+
+- If you were to start this project again, what changes would you make to your approach or the technologies you used?
+
+- How can your solution be extended or modified to handle future requirements, such as adding new microservices or migrating to a different cloud provider?
+
+- What challenges did you encounter during the project, and how did you overcome them?
+
+- How did you ensure that your documentation is clear and complete, and what steps did you take to make it easy for others to understand and maintain your solution?
+
+###### Did the students have a good understanding of the concepts and technologies used in the project?
+
+###### Did the students have the ability to communicate effectively and explain their decisions?
+
+###### Are the students capable to think critically about their solution and consider alternative approaches?
+
+##### Review the Architecture Design.
+
+Review the student's architecture design, ensuring that it meets the project requirements:
+
+1. `Scalability`: Ensure that the architecture can handle varying workloads and can scale up or down as needed.
+
+2. `Availability`: Design the architecture to be fault-tolerant and maintain high availability, even during component failures.
+
+3. `Security`: Incorporate security best practices into the architecture, such as encrypting data at rest and in transit, using private networks, and securing API endpoints.
+
+4. `Cost-effectiveness`: Be mindful of the costs associated with the services and resources you select. Aim to design a cost-effective architecture without compromising performance, security, or scalability.
+
+5. `Simplicity`: Keep the architecture as simple as possible, while still meeting the project requirements. Avoid overcomplicating the design with unnecessary components or services.
+
+###### Did the student Architecture design meets the project requirements?
+
+###### Did the students have the ability to design a cost-effective architecture that meets the project requirements?
+
+###### Did the choice of services and architectural patterns align with best practices for scalability, availability, and security?

```

Comment:
we should add in the readme the links or something that says what are the best practices, otherwise we cant aske it on the audit.

Reply:
then you must explain somewhere that the student will need to explain the best practices to the auditer.

Reply:
fixed